camunda-formio-tasklist-vue/src/components/TaskList.vue

```diff
@@ -798,28 +771,77 @@ export default class Tasklist extends Mixins(TaskListMixin) {
     max: number
   ) {
     this.selectedfilterId = filterId;
-    await CamundaRest.filterTaskListPagination(
+    const paginatedTaskResults = await CamundaRest.filterTaskListPagination(
       this.token,
       filterId,
       requestData,
       first,
       max,
       this.bpmApiUrl
-    ).then((result) => {
-      this.tasks = result.data;
-    });
+    );
+    this.tasks = paginatedTaskResults.data;
+  }
+
+  async onUserSearch (search: string, loading: any) {
+    if(search.length) {
+      loading(true);
+      this.reviewerUsersList = [];
+    }
+
+    if(this.selectSearchType === "firstName") {
```

Comment:
try to create enums for these types of variable `firstName` and then use the enum to compare or assign